pkg/boot/linux.go

```diff
@@ -176,11 +175,11 @@ func (li *LinuxImage) Load(verbose bool) error {
 	defer k.Close()
 
 	// Append device-tree file to the end of initrd
-	if li.DeviceTree != nil {
+	if li.KexecOpts.DTB != nil {
```

Comment:
IIRC KexecOpts is a pointer and you've not yet check whether it is nil. So at the very least this will need some cleaup. Further, this might be a good place to try to break this function into bits that can be test.

Reply:
It is not ;)

```
type LinuxImage struct {
...
   KexecOpts linux.KexecOptions
...
}
```

Reply:
Done.